parser/parser/cc/token.cc

```diff
@@ -29,3 +29,9 @@ std::string_view token::view() const {
 std::string token::asString() const {
     return std::string(_string);
 }
+
+std::ostream &operator<<(std::ostream &o, const ruby_parser::token &token) {
+    return o << "ruby_parser::token{.start = " << token.start() << ", .end = " << token.end()
+             << ", .type = " << ruby_parser::token::tokenTypeName(token.type()) << ", .str = \"" << token.view()
+             << "\" }";
+}
```

Comment:
I opted to implement this using `operator<<` because it doesn't look like we currently use `fmt` anywhere, and I also had to add a `operator<<` implementation for the `ruby_parser::location` type that Bison uses (it *only* lets you use `operator<<` for debug information), so I figured it was nice to be consistent in the `parser/` folder. (I also like that this implementation doesn't materialize an intermediate string).

Reply:
The downside is that it's more annoying to use in `lldb`